src/apartments/__mocks__/repositories/apartment-repository.mock.ts

```diff
@@ -0,0 +1,18 @@
+import { ApartmentRepository } from '../../repositories/apartment.repository';
+import { mockValue } from '../../../common/helpers/mock-value';
+import {
+  deleteResult,
+  updateResult,
+} from '../../../common/__mocks__/base.mock';
+import { apartmentEntity } from '../entities/apartment-entity.mock';
+
+export const ApartmentRepositoryMock = {
+  provide: ApartmentRepository,
+  useValue: {
+    create: mockValue(apartmentEntity),
+    findMany: mockValue([apartmentEntity]),
+    findById: mockValue(apartmentEntity),
+    updateById: mockValue(updateResult),
+    deleteById: mockValue(deleteResult),
```

Comment:
Here you want to have mock implementation for your repository with methods like:

```typescript
findOne: jest.fn((options) => mockItems.find(i => i.id === options.id))
```